### ISSUE TITLE
ZLS causes LSP4IJ to throw UnsupportedOperationException by requesting `workspace/semanticTokens/refresh` 

### ISSUE THREAD

**Issue by fbricon** [redhat-developer/lsp4ij#385]
[zls](https://github.com/zigtools/zls/wiki/Installation)  requests `workspace/semanticTokens/refresh` on startup, which LSP4IJ doesn't support, which  causes this UnsupportedOperationException:

```
Internal error: java.lang.reflect.InvocationTargetException

java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$1(LanguageServerWrapper.java:260)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.UnsupportedOperationException
	at org.eclipse.lsp4j.services.LanguageClient.refreshSemanticTokens(LanguageClient.java:193)
	... 16 more
```

Language Servers should not cause LSP4IJ to crash. We should provide no-op implementations of unsupported features instead.

**Comment by fbricon** [comment]
@FalsePattern have you seen problem that with your semantic highlighting hack?

**Comment by FalsePattern** [comment]
Known issue, the LSP4IJ branch of ZigBrains uses a custom LanguageClient class provided through the language server factory, which does implement that specific method.

Once lsp4ij 0.3.0 is released, I will finish migrating ZigBrains to it and release 16.0.0 with builtin zls support via lsp4ij instead of the custom lsp client.

**Comment by angelozerr** [comment]
This issue will be covered correctly when https://github.com/redhat-developer/lsp4ij/issues/238 will be done.

I have started to implement inmy spare time something by using IJ RainbowVisitor and I can higlight tokens, now I need to manage a better mapping between existing IJ TextAttributes (which manage coloration) and LSP semantic tokens + modifiers.

I'm trying to mimic the same idea than vscode https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#semantic-token-scope-map

**Comment by angelozerr** [comment]
This bug has been fixed since 0.3.0 implements rhis refresh support

**Comment by angelozerr** [comment]
fixed with https://github.com/redhat-developer/lsp4ij/pull/392
